src/lib/data_service.hpp

```diff
@@ -20,7 +23,7 @@ class data_service {
 
     // register a new rpc
     virtual bool bind(std::string const& request_name, std::string const& group_id,
-                      data_service_request_handler_t const& request_cb) = 0;
+                      data_service_request_handler_t const& request_cb, data_service_comp_handler_t const& comp_cb) = 0;
```

Comment:
in server side, comp_handler is optional (defaulted to nullptr), will there be a problem if someone didn't register comp_cb, but here it is mandatory to be non-null?